msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/NativeAuthPublicClientApplicationAbstractTest.kt

```
@@ -98,19 +102,29 @@ abstract class NativeAuthPublicClientApplicationAbstractTest : IPublicClientAppl
         return credential.password
     }
 
-    private fun getConfigsThroughSecretValue(): Map<String, NativeAuthTestConfig.Config>? {
-        val secretValue = KeyVaultFetchHelper.getSecretForBuildAutomation("msalandroidnativeauthautomationconfjsonfile")
+    private fun getConfigsThroughBuildValue(): Map<String, NativeAuthTestConfig.Config>? {
+        // We are no longer fetching the config from the Key Vault, we are using a BuildConfig value
+        // val secretValue = KeyVaultFetchHelper.getSecretForBuildAutomation("msalandroidnativeauthautomationconfjsonfile")
+        var buildConfigString = BuildValues.getNativeAuthConfigString()
+        // If the buildConfigString is null or empty, we will try to read the config from the file path
+        if (buildConfigString == null || buildConfigString.isEmpty()) {
+            val buildConfigFilePath = BuildValues.getNativeAuthConfigFilePath()
+            // If the build config file path is set, read the file and set buildConfigString to its content
+            if (buildConfigFilePath != null && buildConfigFilePath.isNotEmpty()) {
+                buildConfigString = readConfigFile(buildConfigFilePath)
+            }
```

What happen if the build config file path is not set? Can you throw a meaningful error and log what is missing? This will make debug easier.
Something like:
"Native auth config file pipeline variable or local file path not found."

---

Done